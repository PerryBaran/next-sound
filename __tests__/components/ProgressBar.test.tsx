import { fireEvent, render, screen } from "@testing-library/react"
import ProgressBar from "../../src/components/musicPlayer/progressbar/ProgresssBar"

describe("ProgressBar", () => {
  const props = {
    handleAudioTime: jest.fn(),
    time: 10,
    duration: 100
  }

  test("renders correctly", () => {
    render(<ProgressBar {...props} />)
    const input = screen.getByDisplayValue(props.time)

    expect(input).toHaveAttribute("type", "range")
    expect(input).toHaveAttribute("name", "time")
    expect(input).toHaveAttribute("min", "0")
    expect(input).toHaveAttribute("max", `${props.duration}`)
  })

  test("input change", () => {
    render(<ProgressBar {...props} />)

    expect(props.handleAudioTime).toBeCalledTimes(0)

    const newValue = 20
    fireEvent.change(screen.getByDisplayValue(props.time), {
      target: { value: newValue.toString() }
    })

    expect(props.handleAudioTime).toBeCalledTimes(1)
    expect(props.handleAudioTime).toBeCalledWith(newValue)
  })

  test("snapshot", () => {
    const { asFragment } = render(<ProgressBar {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
