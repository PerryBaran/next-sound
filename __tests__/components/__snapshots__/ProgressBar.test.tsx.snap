// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <input
      class="bar"
      max="100"
      min="0"
      name="time"
      type="range"
      value="10"
    />
  </div>
</DocumentFragment>
`;
